### PR TITLE
test: coverage for mersennes31 complex extension

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Currently, adds coverage for `add`, `sub`, `mul` and `mul_2exp_u64`, for complex Mersenne extensions.

### CODE REVIEW

File: mersenne-31/src/complex.rs [hunk 1]
```diff
@@ -255,3 +255,127 @@ impl<AF: AbstractField + AbstractionOf<Mersenne31>> AbstractExtensionField<AF>
...
+            Fi::new_imag(F::TWO) - Fi::new_imag(F::ONE),
+            Fi::new_imag(F::ONE)
+        );
+        assert_eq!(
+            Fi::new_imag(F::NEG_ONE) - Fi::ZERO,
+            Fi::new_imag(F::NEG_ONE)
+        );
+
+        // WARN: panics at the moment, but it should not
+        assert_eq!(
```
Comment (jorgeantonio21) [inline mersenne-31/src/complex.rs:343]: Panics at the moment (but it shouldn't)

**Comment by npwardberkeley** [comment]
Ah does `sub` just need to do a reduction? Yes if you'd like to work on it that'd be great!

File: mersenne-31/src/complex.rs (hunk 1 above)
Comment (jorgeantonio21) [reply mersenne-31/src/complex.rs]: Actually, this was a mistaken statement. After correcting the mistake, it works ! :)

**Review by jorgeantonio21 — COMMENTED** [review]
The previous code version had a bug in the test checks. After correcting it, everything looks fine. The referred modular reduction is performed on equality checks (as it relies on `as_canonical_u32` method).

**Review by dlubarov — APPROVED** [review]
LGTM, thanks!

**Comment by dlubarov** [comment]
I think certain tests could be moved to a generic suite of basic tests that work with any Field, but that can happen later, looks good for now!

**Pull request merged**